examples/cleanup_unwatched_media.py

```diff
@@ -0,0 +1,248 @@
+                hls_resources_to_delete[oid].append(resource_obj["path"])
+                resource_size = resource_obj["file_size"]
+                hls_total_size += resource_size
+    return hls_resources_to_delete, hls_total_size
+
+
+def delete_hls_resources(
+    msc: MediaServerClient, data: list[dict], apply: bool = False
+) -> int:
+    """Delete or simulate deletion of list of hls resources, return deleted count"""
+    hls_resources_to_delete, hls_size = get_hls_resources(data)
+    deleted_resources_count = 0
+    if hls_size:
+        print(
+            f"Cleaning up {len(hls_resources_to_delete)} HLS resources will free "
+            f"{get_human_readable_size(hls_size)}"
+        )
+        for oid, files in hls_resources_to_delete.items():
+            files_str = ",".join(files)
+            params = {"oid": oid, "names": files_str}
+            if apply:
+                print(f"Deleting resources of oid {oid}: {files}")
+                r = msc.api("/medias/resources-delete/", method="post", data=params)
+                if not r["success"]:
+                    print(f"Error when deleting {oid}: {r['message']}")
+                else:
+                    deleted_resources_count += len(files)
+            else:
+                print(f"[Dry Run] Would delete resources of oid {oid}: {files}")
+        if apply:
+            print(f"Deleted {deleted_resources_count} resources")
+        else:
+            print(
+                f"[Dry run] Could have freed up to {get_human_readable_size(hls_size)} by deleting HLS resources"
+            )
+    else:
+        print("No HLS resources to cleanup")
+    return deleted_resources_count
+
+
+def delete_unwatched_media(
+    msc: MediaServerClient, data: list[dict], apply: bool = False
+) -> [int, str]:
+    """Delete or simulate deletion of unwatched media, return deleted count and
+    log file path which can be used to revert the operation using the mass_untrash script.
+    Note that this function is really dangerous if the platform does not have the trash feature enabled
+    """
+    trashed_media_count = 0
+    trashed_files_log_path = None
+    media_size = get_total_size(data)
+    if media_size:
+        media_to_delete_count = len(data)
+        print(
+            f"Trashing {media_to_delete_count} unwatched media will free {get_human_readable_size(media_size)}"
+        )
+        oids = get_oids(data)
+        params = {"oids": oids}
+        if apply:
+            trashed_files_log_path = f"{time.strftime('%Y%m%d-%H%M%S')}-trashed.csv"
+            print(f"Will put {media_to_delete_count} media to trash")
+            deleted_statuses = msc.api(
+                "/catalog/bulk_delete/", method="post", data=params
+            )["statuses"]
+            trashed_media_count = 0
+            for deleted_status in deleted_statuses.values():
+                if deleted_status["status"] == 200:
+                    trashed_media_count += 1
```

Comment:
`trashed_media_count = sum(1 for st in deleted_statuses.values() if st["status"] == 200)`